Update the Hacktoberfest landing before the event starts (#80)
gradle/community: docs/events/hacktoberfest/2024/README.md

[Hacktoberfest](https://hacktoberfest.com/) is a month-long celebration of open-source projects, their maintainers and contributors.
Many component maintainers from the Gradle ecosystem, such as the
[Gradle plugin for Jenkins](https://plugins.jenkins.io/gradle),
already include their projects in Hacktoberfest.

This year, Gradle is participating in Hacktoberfest!
We are looking forward to collaborating with all contributors on the featured projects, hosting online events for new open-source contributors,
and sending custom Gradle goodies for those who do substantial contributions to the Gradle ecosystem.

## Quick Start

1. Sign-up for Hacktoberfest on [hacktoberfest.com](https://hacktoberfest.com/)
2. Sign-up in [this form for Gradle](https://forms.gle/go2VESj7kDG1QUqV7) so that
  we can contact you and distribute swag
3. Join the `#hacktoberfest` channel on the Gradle Community Slack [](../../../contributing/community-slack.md)
4. Check out the Featured Projects below, and consider joining us for Hacktoberfest events!
5. Enjoy!

## Featured Projects

During Hacktoberfest,

> you are welcome to work on any project marked with `hacktoberfest` topic on GitHub or Gitlab.

Reviewer: ```suggestion
you are welcome to work on any project marked with the `hacktoberfest` topic on GitHub or Gitlab.
```

> Of course, invite you to work on the Gradle related projects:

Reviewer: ```suggestion
Of course, we invite you to work on the following Gradle related projects:
```

plugins, tools, documentation and examples.

> Below, there are a few projects we have in mind for the event,

Reviewer: ```suggestion
Below, there are a few project ideas we have in mind for the event,
```

> with the different levels of complexity:

Reviewer: ```suggestion
with different levels of complexity:
```

- **Beginner:** [Learn Gradle](#learn) - Basics, Java, Kotlin
- **Easy:** [Adopt Gradle Best Practices in your GitHub Actions](#github-actions) -
  GitHub Actions, Scripting, Documentation
- **Advanced:** [Configuration Cache Compatibility in Plugins](#configuration-cache) -
  Java, Kotlin, Gradle
- **Easy/Medium:** [Expand the Gradle Cookbook](#cookbook) -
  Documentation, Markdown, tech of your choice
- **Medium:** [Contribute to GradleUp Plugins](#gradleup) -
  Java, Kotlin, Android, Documentation
- **Easy/Medium:** [Improve this Site!](#community-site) -
  MkDocs, Web Design, Documentation

> 

Reviewer: It would be cool if the list above was in order of difficulty.

Author: Will update in the next batch

<a name="github-actions"></a>

### Adopt Gradle Best Practices in your GitHub Actions

> Use Gradle in **YOUR** GitHub repositories and use GitHub Actions to build projects?

Reviewer: This sentence should be removed.
```suggestion
```

> We invite everyone to adopt the [Gradle & GitHub Actions best practices](https://community.gradle.org/cookbook/ci/github-actions/) in them.

Reviewer: ```suggestion
We invite everyone to adopt [Gradle & GitHub Actions best practices](https://community.gradle.org/cookbook/ci/github-actions/).
```

You can:

* Make them safer by enabling the [Dependency Submission Action](https://github.com/gradle/actions/blob/main/dependency-submission/README.md) to [detect vulnerable dependencies with](https://community.gradle.org/cookbook/ci/github-actions/#detect-vulnerable-dependencies-with-a-dependency-submission-workflow) -
  [How To](https://community.gradle.org/cookbook/ci/github-actions/#detect-vulnerable-dependencies-with-a-dependency-submission-workflow)
* Adopt the [`setup-gradle` action](https://community.gradle.org/cookbook/ci/github-actions/) to enable [Build Caching in your builds](https://community.gradle.org/cookbook/ci/github-actions/#enable-caching-of-downloaded-artifacts)
* Connect them to [Gradle Build Scan](https://docs.gradle.org/current/userguide/build_scans.html) to get better insights and troubleshooting experience - [How To](https://community.gradle.org/cookbook/ci/github-actions/#configure-github-actions)
* Update documentation for the Actions, based on your experiences
  ([GitHub Repo](https://github.com/gradle/actions), [Cookbook Pages](https://community.gradle.org/cookbook/ci/github-actions/))

> To note your patches within the Hacktoberfest,

Reviewer: ```suggestion
To tag your patches for Hacktoberfest,
```

please reference [this ticket](https://github.com/gradle/actions/issues/406)

> in the pull requests toward your repository,

Reviewer: ```suggestion
in the pull requests,
```

> which should also have the `hacktoberfest` topic.

Reviewer: ```suggestion
and add the `hacktoberfest` topic.
```

References:

- [GitHub Repository](https://github.com/gradle/actions)
- Discuss: `#github-integrations` channel on the Community Slack

<a name="configuration-cache"></a>

### Configuration Cache Compatibility in Plugins

> In the upcoming Gradle 9.0, we plan to recommend enabling [Gradle Configuration Cache](https://docs.gradle.org/current/userguide/configuration_cache.html) by default.

Reviewer: ```suggestion
In the upcoming Gradle 9.0, we plan to recommend enabling the [Gradle Configuration Cache](https://docs.gradle.org/current/userguide/configuration_cache.html) by default.
```

for cases that are not included in the main Gradle documentation,
including tool integrations, solutions for specific languages and tech stacks
(e.g. Android), tips on troubleshooting and plugin development.

You are welcome to create new recipes, or to update the existing ones.
For example, we know that some CI recipes are dated, and contributions are needed!

References:

- [Contributing to the Gradle Cookbook](https://community.gradle.org/cookbook/CONTRIBUTING/)
- [Open Issues](https://github.com/gradle/community/issues?q=is%3Aissue+is%3Aopen+label%3Acookbook)
- Discuss: `#docs` channel on the Community Slack

<a name="gradleup"></a>

### Contribute to GradleUp Plugins

[GradleUp](https://gradleup.com/) is a new GitHub organization focusing on simplifying Java, Kotlin and Android development with Gradle.
The team nominated a few projects for Hacktoberfest: Shadow Plugin, Static Analysis Plugin, Gratatouille, etc. Contributions to other projects are welcome, too!

References:

- [Contributing to GradleUp](https://gradleup.com/docs/community/participate/)
- Discuss: `#gradleup` channel on the Community Slack

<a name="learn"></a>

### Learn Gradle

> If you are just starting with Gradle,

Reviewer: ```suggestion
If you are just starting out with Gradle,
```

> maybe you could start from learning it and practicing with common

Reviewer: ```suggestion
you can learn through common
```

Java, Kotlin or Android projects.

* Learn Gradle Build Tool with free courses on [DPE University](https://dpeuniversity.gradle.com/app/catalog). Learn Gradle features, create new repositories while learning, and make sure to mark them with `#hacktoberfest`!
* If you see any issues with the documentation or learning materials,
  please submit issues or pull requests.

More advanced users can also experiment with [Declarative Gradle](https://declarative.gradle.org/) and share their feedback.
See the [Getting Started Guide](https://declarative.gradle.org/docs/getting-started/) for the pointers.

> To discuss your learning experiences, use the `#docs` channel on the Community Slack

Reviewer: ```suggestion
To discuss your learning experiences, use the `#docs` channel on the Community Slack.
```

<a name="community-site"></a>

### Improve the Community Site

We created this community site in early 2024, and there are many improvements to be done!

> Improving look&feel, creating new content, etc., etc.

Reviewer: ```suggestion
Improving the look & feel, creating new content, etc...
```

All patches are welcome!

References:

* [Contributing to the Community Site](./../../../CONTRIBUTING.md)
* [Open Issues](https://github.com/gradle/community/labels/website)

## ¿More Featured Projects?

During the Hacktoberfest, maintainers are welcome to add their project if they provide contributor guidelines and issues that can be worked on by contributors.
If you want to add your project, add a `hacktoberfest` topic to it and submit a pull request to this page.

> It count as contribution, too :wink:

Reviewer: ```suggestion
It counts as a contribution, too :wink:
```

## Swag

In addition to our Hacktoberfest swag,
we plan to send something nice to those who make substantial contributions to
Gradle Build Tool and its ecosystem.

> For that, make sure to register in [Our Sign-Up Form](https://forms.gle/go2VESj7kDG1QUqV7) so that we can discover your contributions.

Reviewer: ```suggestion
Make sure to register in [Our Sign-Up Form](https://forms.gle/go2VESj7kDG1QUqV7) so that we can discover your contributions.
```

![Hacktoberfest Swag](./images/swag.jpg)

## Events

We plan some online events during October, with a goal to help with contributions.

> Openings of those events will be recorded

Reviewer: ```suggestion
Those events will be recorded:
```
